### PR TITLE
complete api doc

### PR DESCRIPTION
**Pull request by ramsayleung**
Complete unfinished API doc.

### CODE REVIEW

**Review by davis999 — COMMENTED** [review]
1. add Numbering.
2. all update action name is wrong, like "set_attribute_label".
3. some table confuse `field name` and `field type`.
4. Boolean field's name should not start with "is".

**Comment by ramsayleung** [comment]
1. add Numbering.
2. modify name in update action.
3. fix issue that confuses `field name` and `field type`
4. I know Boolean field's name should not start with "is", but I write the document  according to code, this is how the code defines a Boolean field. I have no choice If I don't modify code.

File: docs/api.md [hunk 1]
```diff
@@ -28,32 +27,208 @@ TODO
 | isSearchable | Boolean | required |
```
Comment (davis999) [inline docs/api.md]: Boolean field's name should not start with "is".

File: docs/api.md [hunk 2]
```diff
@@ -28,32 +27,208 @@ TODO
...
 
-### ProductTypeView
+### 2.3. ProductTypeView
 
-## API
+| field name | field type | comments |
+|-----|------|-----|
+| id | String | |
+| version | Integer | |
+| createAt | ZonedDateTime | |
```
Comment (davis999) [inline docs/api.md]: should be "createdAt".

File: docs/api.md [hunk 3]
```diff
@@ -28,32 +27,208 @@ TODO
...
+| attributes | List\<AttributeDefinitionView\> | |
+
+### 2.4. AttributeDefinitionView
+| field name | field type | comments |
+|-----|------|-----|
+| type | AttributeType | |
+| name | String | |
+| label | LocalizedString | |
+| inputTip | LocalizedString | |
+| isRequired | Boolean | |
```
Comment (davis999) [inline docs/api.md]: Boolean field's name should not start with "is".

File: docs/api.md [hunk 4]
```diff
@@ -28,32 +27,208 @@ TODO
...
+### 2.3. ProductTypeView
 
-## API
+| field name | field type | comments |
+|-----|------|-----|
+| id | String | |
+| version | Integer | |
+| createAt | ZonedDateTime | |
+| lastModifiedAt | ZonedDateTime | |
+| key | String | |
```
Comment (davis999) [inline docs/api.md]: remove "key".

File: docs/api.md [hunk 5]
```diff
@@ -28,32 +27,208 @@ TODO
...
+| name | LocalizedString | required, NotNull |
+
+#### 2.6.14. SetLocalizedEnumValueLabel
+
+| field name | field type | comments |
+|-|-|-|
+| action | String | required, set as `setLocalizedEnumValueLabel` |
+| name | LocalizedString | required, NotNull |
+
+#### 2.6.15. SetIsSearchable
```
Comment (davis999) [inline docs/api.md]: should be "SetSearchable".

File: docs/api.md [hunk 6]
```diff
@@ -28,32 +27,208 @@ TODO
...
+| field name | field type | comments |
+|-----|------|-----|
+| type | AttributeType | |
+| name | String | |
+| label | LocalizedString | |
+| inputTip | LocalizedString | |
+| isRequired | Boolean | |
+| attributeConstraint | AttributeConstraint | |
+| inputHint | TextInputHint | |
+| isSearchable | Boolean | |
```
Comment (davis999) [inline docs/api.md]: Boolean field's name should not start with "is".

File: docs/api.md [hunk 7]
```diff
@@ -28,32 +27,208 @@ TODO
...
+### 2.5. UpdateRequest
+
+| field name | field type | comments |
+|-|-|-|
+| version | Integer | required, NotNull, min is 0 |
+| actions | List\<UpdateAction\> | required, NotNull |
+
+### 2.6. UpdateAction
+
+#### 2.6.1. SetKey
```
Comment (davis999) [inline docs/api.md]: remove this action.

File: docs/api.md (hunk 1 above)
Comment (ramsayleung) [reply docs/api.md]: attributedefinitiondraft java - product-type - dev product-type 2017-04-26 18-02-57

File: docs/api.md (hunk 3 above)
Comment (ramsayleung) [reply docs/api.md]: attributedefinitiondraft java - product-type - dev product-type 2017-04-26 18-04-38

File: docs/api.md (hunk 6 above)
Comment (ramsayleung) [reply docs/api.md]: attributedefinitiondraft java - product-type - dev product-type 2017-04-26 18-02-57

File: docs/api.md (hunk 5 above)
Comment (ramsayleung) [reply docs/api.md]: updateaction java - product-type - dev product-type 2017-04-26 18-05-49

File: docs/api.md (hunk 1 above)
Comment (davis999) [reply docs/api.md]: It is our rule: Don't use isXXX as a boolean filed, use adjective.

File: docs/api.md (hunk 3 above)
Comment (davis999) [reply docs/api.md]: It is our rule: Don't use isXXX as a boolean filed, use adjective.

File: docs/api.md (hunk 6 above)
Comment (davis999) [reply docs/api.md]: It is our rule: Don't use isXXX as a boolean filed, use adjective.

File: docs/api.md (hunk 5 above)
Comment (davis999) [reply docs/api.md]: It is our rule: Don't use isXXX as a boolean filed, use adjective.

File: docs/api.md (hunk 6 above)
Comment (ramsayleung) [reply docs/api.md]: Of course I know the rule:
>3.4 boolean field

>Don't use isXXX as a boolean filed, use adjective. 

But I am talking about the code

File: docs/api.md (hunk 5 above)
Comment (ramsayleung) [reply docs/api.md]: change to "setSearchable".

File: docs/api.md (hunk 7 above)
Comment (ramsayleung) [reply docs/api.md]: removed.

File: docs/api.md (hunk 6 above)
Comment (ramsayleung) [reply docs/api.md]: change to "searchable".

File: docs/api.md (hunk 3 above)
Comment (ramsayleung) [reply docs/api.md]: change to requirable.

File: docs/api.md (hunk 4 above)
Comment (ramsayleung) [reply docs/api.md]: removed "key"

File: docs/api.md (hunk 2 above)
Comment (ramsayleung) [reply docs/api.md]: change to "createdAt"

File: docs/api.md (hunk 1 above)
Comment (ramsayleung) [reply docs/api.md]: change to "searchable".

File: docs/api.md [hunk 8]
```diff
@@ -1,59 +1,226 @@
...
+| attributes | List\<AttributeDefinitionView\> | |
+
+### 2.4. AttributeDefinitionView
+| field name | field type | comments |
+|-----|------|-----|
+| type | AttributeType | |
+| name | String | |
+| label | LocalizedString | |
+| inputTip | LocalizedString | |
+| requirable | Boolean | |
```
Comment (davis999) [inline docs/api.md]: should be "required".

File: docs/api.md [hunk 9]
```diff
@@ -1,59 +1,226 @@
...
-### AttributeDefinitionDraft
+### 2.2. AttributeDefinitionDraft
 
 | field name | field type | comments |
 |-----|------|-----|
 | type | AttributeType | required |
 | name | String | required, The name must be between two and 36 characters long and can contain the ASCII letters A to Z in lowercase or uppercase, digits, underscores (_) and the hyphen-minus (-) |
 | label | LocalizedString | required |
-| isRequired | Boolean | required |
+| requirable | Boolean | required |
```
Comment (davis999) [inline docs/api.md]: should be "required".

Comment (ramsayleung) [reply docs/api.md]: change to "required".

File: docs/api.md (hunk 8 above)
Comment (ramsayleung) [reply docs/api.md]: change to required.

**Comment by ramsayleung** [comment]
1. fix the issue that misunderstand updateAction file type and  file name
2. add plainEnumValue and LocalizedEnumValue.

**Pull request merged**